### PR TITLE
seperated data loaders classes

### PR DESCRIPTION
**Pull request by hardlyhuman**
This PR is solving issue #7 by refactoring the data_utils.py file to dynamically load only the necessary data loader libraries instead of importing all of them upfront. This change prevents the need to load all the libraries in memory thus reducing unnecessary dependencies.

Changes:

- Seperated Data Loader Classes
- Implemented dynamic imports for data loaders
- Updated the organization of data_utils.py to align with dynamic import strategy.

### CODE REVIEW

File: src/utils/data_utils.py [hunk 1]
```diff
@@ -448,7 +109,7 @@ def extr_noniid(train_dataset, samples_per_client, classes):
     return Subset(all_data, perm[:samples_per_client])
 
 
-def cifar_extr_noniid(
+def  cifar_extr_noniid(
```
Comment (tremblerz) [inline src/utils/data_utils.py:112]: just a minor thing

**Review by tremblerz — APPROVED** [review]
Thanks, this is great! Can you confirm, the code was running after the changes?

**Comment by hardlyhuman** [comment]
Tested with Cifar10 for 2 rounds of FL and it was successful. Would you recommend to test with all other datasets as well once?

Also, I have reduced the training set size drastically since I am runing it on my PC and primarily testing the functionality rather than ML accuracy part.

**Comment by tremblerz** [comment]
should be good enough, thanks!
